src/plots/cartesian/axes.js

```diff
@@ -378,7 +378,11 @@ axes.saveRangeInitial = function(gd, overwrite) {
 //      tozero: (boolean) make sure to include zero if axis is linear,
 //          and make it a tight bound if possible
 axes.expand = function(ax, data, options) {
-    var needsAutorange = (ax.autorange || Lib.nestedProperty(ax, 'rangeslider.autorange'));
+    var needsAutorange = (
+        ax.autorange ||
+        !!Lib.nestedProperty(ax, 'rangeslider.autorange').get()
```

Comment:
Even if `ax.rangeslider` is undefined, `Lib.nestedProperty(ax, 'rangeslider.autorange')` return an object (which is of course truthy) - which made `needsAutorange` always be truthy, removing the early return :racehorse: path.